File: spec/go/debug_array_user_eof_exception_test.go
package spec

import (
	"runtime/debug"
	"os"
	"testing"
	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	. "test_formats"
	"github.com/stretchr/testify/assert"
	"io"
)

func TestDebugArrayUserEofException(t *testing.T) {
	defer func() {
		if r := recover(); r != nil {
			debug.PrintStack()
			t.Fatal("unexpected panic:", r)
		}
	}()
	f, err := os.Open("../../src/nav_parent_codes.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)
	var r DebugArrayUserEofException
	err = r.Read(s, &r, &r)
	assert.Error(t, err)
	assert.ErrorIs(t, err, io.EOF)

	assert.EqualValues(t, 3, r.OneCat.Meow)
	assert.EqualValues(t, 73, r.OneCat.Chirp)
	assert.EqualValues(t, 3, len(r.ArrayOfCats))
	assert.EqualValues(t, 49, r.ArrayOfCats[0].Meow)
	assert.EqualValues(t, 50, r.ArrayOfCats[0].Chirp)
	assert.EqualValues(t, 51, r.ArrayOfCats[1].Meow)
	assert.EqualValues(t, 66, r.ArrayOfCats[1].Chirp)
	assert.EqualValues(t, 98, r.ArrayOfCats[2].Meow)
}

File: spec/php/DebugArrayUserEofExceptionTest.php
<?php
namespace Kaitai\Struct\Tests;

class DebugArrayUserEofExceptionTest extends TestCase {
    public function testDebugArrayUserEofException() {
        $r = DebugArrayUserEofException::fromFile(self::SRC_DIR_PATH . '/nav_parent_codes.bin');

        try {
            // --debug implies --no-auto-read
            $r->_read();
            $this->fail("An exception was expected, but none thrown");
        } catch (\Kaitai\Struct\Error\EndOfStreamError $e) {
            // OK
        }

        $this->assertSame(3, $r->oneCat()->meow());
        $this->assertSame(73, $r->oneCat()->chirp());
        $this->assertSame(3, count($r->arrayOfCats()));
        $this->assertSame(49, $r->arrayOfCats()[0]->meow());
        $this->assertSame(50, $r->arrayOfCats()[0]->chirp());
        $this->assertSame(51, $r->arrayOfCats()[1]->meow());
        $this->assertSame(66, $r->arrayOfCats()[1]->chirp());
        $this->assertSame(98, $r->arrayOfCats()[2]->meow());
    }
}
